### ISSUE TITLE
cpplint output error when it is a git submodule

### ISSUE THREAD

**Issue by Wsine** [cpplint/cpplint#37]
cpplint checks a project and prints no error found. 
When the same project is as a git submodule, cpplint prints multiple errors.

```
include/base_code.hpp:6:  #ifndef header guard has wrong style, please use: COMPANY_COMMON_INCLUDE_BASE_CODE_HPP_  [build/header_guard] [5]
include/base_code.hpp:56:  #endif line should be "#endif  // COMPANY_COMMON_INCLUDE_BASE_CODE_HPP_"  [build/header_guard] [5]
```

**Comment by rjmunro** [comment]
What are the exact commands you are typing to run cpplint?

Can you copy the lines in the source files that are causing the issues, preferably with a few lines before and after for context?

**Comment by Wsine** [comment]
I use `# cpplint --recursive include/*` to check my codes.
I found it is the header guard problem. using `--repository=.` to specify the top level directory of the repository can avoid this problem. but why it cant detect current directory as the top level directory?

**Comment by Durant35** [comment]
@Wsine your option `--repository=.` works well for me
